src/GridView/CardGrid.js

```diff
@@ -28,6 +28,43 @@ class CardGrid extends ConnectionComponent {
 		}
 	}
 
+	isCoveredByFilter(childEntity) {
+		// every is equivalent to logical and over an array
+		return this.props.filter.every((filter) => {
+			return this.testFilter(childEntity, filter);
+		});
+	}
+
+	testFilter(childEntity, filter) {
+		if (!filter.value) {
+		 	return true;
+		 }
+		 const attributeValues = this.getAttributeValues(childEntity);
+
+		if (this.doesContain(childEntity.Name, filter.value)){
+			return true;
+		}
+		else if (this.doesContain(attributeValues, filter.value)){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either merge this branch with the identical one on line "44" or change one of the implementations. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1871)